### PR TITLE
fix: add option to deduplicate tables after backfill

### LINKED ISSUE #300: Feat: Optionally run optimize commands after backfill
When running a big backfill it's possible that duplicate events exist. Running the optimize final commands will fix it, but can take a very long time on huge datasets. We should make it an option to run those commands immediately afterbackfill, though. They should be run on all xapi schema tables that are `ReplacingMergeTree`.

### PR DESCRIPTION
**Pull request by Ian2012**
## Description

This PR adds a `deduplicate` option for the `transform_tracking_log` command that will run the `OPTIMIZE table FINAL` for every table that needs it.

Fixes #300

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @Ian2012!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


This is currently a draft pull request. When it is ready for our review and all tests are green, click "Ready for Review", or remove "WIP" from the title, as appropriate.

File: README.rst [hunk 1]
```diff
@@ -70,7 +70,10 @@ Usage
 
 #. (Optional) Sink Historical event data to ClickHouse::
 
-    tutor [dev|local] run lms ./manage.py lms transform_tracking_logs --source_provider LOCAL --source_config '{"key": "/openedx/data", "container": "logs", "prefix": "tracking.log"}' --transformer_type xapi
+    tutor [dev|local] do transform_tracking_logs \
+      --source_provider LOCAL --source_config '{"key": "/openedx/data", "container": "logs", "prefix": "tracking.log"}' \
+      --transformer_type xapi --deduplicate
+
```
Comment (bmtcril) [inline README.rst:76]: Can we add another note here about situations where you would need to deduplicate and why you shouldn't do that if you don't need to (takes a long time).

File: tutoraspects/commands_v1.py [hunk 2]
```diff
@@ -165,7 +167,16 @@ def dump_courses_to_clickhouse(options) -> list[tuple[str, str]]:
...
     ),
 )
-def transform_tracking_logs(**kwargs) -> list[tuple[str, str]]:
+@click.option(
+    "--deduplicate",
+    is_flag=True,
+    help=(
+        "A flag to determine if the command should run optimize the tables for deduplication."
+        "If present, the OPTIMIZE FINAL query will be applied for all tables that have a "
+        "deduplication strategy."
```
Comment (bmtcril) [inline tutoraspects/commands_v1.py:176]: Can we add something like: "This should only be added if you believe events will be duplicated such as replaying logs that have already been added. De-duplication can take a very long time to process."

**Review by bmtcril — APPROVED** [review]

**Comment by openedx-webhooks** [comment]
@Ian2012 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/tutor-contrib-aspects&entry.867055334=https://github.com/openedx/tutor-contrib-aspects/pull/323&entry.1484655318=https://github.com/Ian2012&entry.752974735=2023-08-25+19:03&entry.1917517419=2023-08-28+16:13&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**